.gitignore

```diff
@@ -3,6 +3,7 @@ __pycache__
 .gradle
 .venv
 application-dev*.yaml
+!application-dev*.yaml.sample
```

Comment:
Oh, interesting! I would have thought the `.sample` suffix would have caused the rule on the previous line to not match.